libgnucash/app-utils/gnc-ui-util.c

```diff
@@ -1863,6 +1863,39 @@ xaccPrintAmount (gnc_numeric val, GNCPrintAmountInfo info)
     return buf;
 }
 
+const char *
+gnc_print_amount_with_bidi_ltr_isolate (gnc_numeric val, GNCPrintAmountInfo info)
+{
+    /* hack alert -- this is not thread safe ... */
+    static char buf[1024];
+
+    if (!xaccSPrintAmount (buf, val, info))
+        buf[0] = '\0';
+
+    if (buf[0] && info.use_symbol)
+    {
+        char temp_buf[1024];
```

Comment:
This is a bit of a problem because it needs 6 bytes more than `buf` to accommodate the two bidi code points. Maybe a `strlen` check to make sure that there's enough left over or use `strncat (temp_buf, buf, buflen - 7)` to ensure that `temp_buf` doesn't get overrun?

Since this gets called a lot maybe a bit of hackery would be worthwhile:
```
#define BUFLEN 1024
static char buf[BUFLEN];
static const char[] ltr_isolate = { 0xe2, 0x81, 0xa6 };
static const char[] ltr_pop_isolate = { 0xe2, 0x81, 0xa9 };
size_t offset = info.use_symbol ? 3 : 0;

memset (buf, 0, BUFLEN);
if (!xaccSPrintAmount (buf + offset, val, info))
{
    buf[0] = '\0';
    return buf;
};

if (!info.use_symbol)
    return buf;

memcpy (buf, ltr_isolate, 3);
if (buf[BUFLEN] != '\0')
{
    buf[BUFLEN] = '\0';
    memcpy (buflen + BUFLEN - 4, ltr_pop_isolate, 3);
}
else
{
    size_t length = strlen (buf);
    memcpy (buflen + length, ltr_pop_isolate, 3);
}

return buf;
```

Reply:
Don't think I have used those mem functions, needed to look them up and seems a better approach, needed to tweak it slightly to get to work.